enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,829 @@
+the destination cluster. Hence, the neighbor caches (ARP, NDP) will be updated.
+This means there's nothing special we need to do on this regard - other than
+ensuring the gateway MAC address is not already available in the logical network.
+
+This flow is described in more detail (and presents alternatives to it) in the
+[OVN-Kubernetes enhancement proposal](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5238).
+
+### Workflow Description
+
+#### Roles
+- admin: a user with cluster admin rights. They can list NADs in all
+  namespaces, create cluster UDNs, and create / update `IPPool`s.
+- VM owner: a user without cluster admin rights. They own their namespaces.
+  They will either import - or create - a VM in the namespace they own.
+
+#### Sequence diagram for the proposed workflow
+```mermaid
+sequenceDiagram
+  actor Admin
+  actor VM Owner
+
+  participant MTV
+  participant CNV
+  participant o as OVN-Kubernetes
+
+  Admin ->> o: provision cluster UDN
+  o -->> Admin: OK
+
+  VM Owner ->> MTV: import VM
+  MTV ->> CNV: create VM(primaryUDNMac=origMAC, primaryUDNIPs=...)
+  CNV ->> o: create pod(mac=origMAC, IPs=ips)
+  o -->> CNV: OK
+  CNV -->> MTV: OK
+  MTV -->> VM Owner: OK
+```
+
+### API Extensions
+
+#### IPAMClaim CRD
+
+The IPAMClaim CRD status sub-resource will need to be updated, adding
+conditions.
+
+For traceability, we also suggest adding to the `IPAMClaim` spec an attribute
+to indicate which pod is holding the claim at any given time. On a VM live
+migration, the OVN-Kubernetes control plane would update the `OwnerPod` after
+the claim has been consumed by a different pod. Same for VM start/stop
+scenarios.
+
+If we choose the [de-centralized IP management](#de-centralized-ip-management)
+alternative, we will also need to change the IPAMClaim CRD spec, adding it an
+attribute to request specific IPs for the workload which will consume the
+claim.
+
+Below you'll see the changes requested:
+```go
+type IPAMClaimSpec struct {
+	// The network name for which this persistent allocation was created
+	Network string `json:"network"`
+	// The pod interface name for which this allocation was created
+	Interface string `json:"interface"`
++	// The IPs requested by the user
++	// +optional
++	IPRequests []CIDR           `json:"ipRequests,omitempty"`
++	MACRequest net.HardwareAddr `json:"macRequest,omitempty"`
+}
+
+// IPAMClaimStatus contains the observed status of the IPAMClaim.
+type IPAMClaimStatus struct {
+    // The list of IP addresses (v4, v6) that were allocated for the pod interface
+-   IPs []string `json:"ips"`
++   IPs []CIDR `json:"ips"`
++   // The name of the pod holding the IPAMClaim
++   OwnerPod string `json:"ownerPod"`
++   // Conditions contains details for one aspect of the current state of this API Resource
++   Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+The `IPAMClaim` status will have (at least) the following conditions:
```

Comment:
Are we also going to denote mac collision at the IPAMClaim status ? is so we have to define it here.

Reply:
Hm ... didn't think of that. 

It would be useful, at least for persisting the error ... 

Still, I'd rather keep anything MAC related out of the IPAMClaim. 

Meaning, the answer is no - unless we come up w/ compelling reasons to do it.